frontend/src/component/common/DialogFormTemplate/DialogFormTemplate.tsx

```diff
@@ -0,0 +1,112 @@
+    createButtonProps: IPermissionButtonProps;
+    description: string;
+    errors: { [key: string]: string };
+    handleSubmit: FormEventHandler<HTMLFormElement>;
+    icon: React.ComponentType;
+    Limit?: React.ReactNode;
+    name: string;
+    onClose: () => void;
+    configButtons: React.ReactNode;
+    resource: string;
+    setDescription: (newDescription: string) => void;
+    setName: (newName: string) => void;
+    validateName?: () => void;
+    testIds?: Partial<{
+        nameInput: string;
+        descriptionInput: string;
+        submitButton: string;
+    }>;
+};
+
+export const DialogFormTemplate: React.FC<FormProps> = ({
+    Limit,
+    handleSubmit,
+    name,
+    setName,
+    description,
+    setDescription,
+    errors,
+    icon,
+    resource,
+    onClose,
+    configButtons,
+    createButtonProps,
+    validateName = () => {},
+}) => {
+    const StyledIcon = styleIcon(icon);
+
+    return (
+        <StyledForm onSubmit={handleSubmit}>
+            <TopGrid>
+                <StyledIcon aria-hidden='true' />
+                <StyledHeader variant='h2'>Create {resource}</StyledHeader>
+                <ProjectNameContainer>
+                    <StyledInput
+                        label={`${resource} name`}
+                        aria-required
+                        value={name}
+                        onChange={(e) => setName(e.target.value)}
+                        error={Boolean(errors.name)}
+                        errorText={errors.name}
+                        onBlur={validateName}
+                        onFocus={() => {
+                            delete errors.name;
+                        }}
+                        autoFocus
+                        InputProps={{
+                            style: { fontSize: theme.typography.h1.fontSize },
+                        }}
+                        InputLabelProps={{
+                            style: { fontSize: theme.typography.h1.fontSize },
+                        }}
+                        size='medium'
+                    />
+                </ProjectNameContainer>
+                <ProjectDescriptionContainer>
+                    <StyledInput
+                        size='medium'
+                        className='description'
+                        label='Description (optional)'
+                        multiline
+                        maxRows={3}
+                        value={description}
+                        onChange={(e) => setDescription(e.target.value)}
+                        InputProps={{
+                            style: { fontSize: theme.typography.h2.fontSize },
+                        }}
+                        InputLabelProps={{
+                            style: { fontSize: theme.typography.h2.fontSize },
+                        }}
+                    />
+                </ProjectDescriptionContainer>
+            </TopGrid>
+
+            <ConfigButtons>{configButtons}</ConfigButtons>
```

Comment:
Discussion point: should we instead use children for the config buttons or is an explicit prop better?

Reply:
I think using prop is fine for now, because it is not clear what the children must be. Are they input fields or buttons or something completely different.